d2/src/main/java/com/linkedin/d2/balancer/simple/SimpleLoadBalancer.java

```diff
@@ -1161,14 +1238,33 @@ public int getClusterCount(String clusterName, String scheme, int partitionId) t
     {
       return clusterCountFutureCallback.get(_timeout, _unit);
     }
-    catch (ExecutionException | TimeoutException | IllegalStateException | InterruptedException e )
+    catch (ExecutionException | TimeoutException | IllegalStateException | InterruptedException e)
     {
+      if (e instanceof TimeoutException)
+      {
+        return getClusterCountFromCache(clusterName, scheme, partitionId);
+      }
       die("ClusterInfo", "PEGA_1017, unable to retrieve cluster count for cluster: " + clusterName +
           ", scheme: " + scheme + ", partition: " + partitionId + ", exception: " + e);
       return -1;
     }
```

Comment:
Nit, it's much clearer for readers to avoid increasing the nest level with `if` blocks, especially if you can trivially avoid it like you can here:
```suggestion
    catch (TimeoutExeption e)
    {
      return getClusterCountFromCache(clusterName, scheme, partitionId);
    }
    catch (ExecutionException | IllegalStateException | InterruptedException e)
    {
      die("ClusterInfo", "PEGA_1017, unable to retrieve cluster count for cluster: " + clusterName +
          ", scheme: " + scheme + ", partition: " + partitionId + ", exception: " + e);
      return -1;
    }
```

Reply:
make sense. Done

Reply:
I don't understand what you mean by this. My suggestion is semantically equivalent to the code you have here, it's just cleaner. The returned value from `getClusterCountFromCache` isn't checked here

Reply:
HI, @PapaCharlie 
 According to my latest code. What I mean here is that: If we extract `TimeoutExeption` to another catch.  The code will be changed to following,  which may be unclean.
```java
    try
    {
      return clusterCountFutureCallback.get(_timeout, _unit);
    }
    catch (TimeoutException e)
    {
      int clusterCount = getClusterCountFromCache(clusterName, scheme, partitionId);
      if (clusterCount >= 0)
      {
        return clusterCount;
      }
      die("ClusterInfo", "PEGA_1017, unable to retrieve cluster count for cluster: " + clusterName +
              ", scheme: " + scheme + ", partition: " + partitionId + ", exception: " + e);
      return -1;
    }
    catch (ExecutionException | IllegalStateException | InterruptedException e)
    {
      die("ClusterInfo", "PEGA_1017, unable to retrieve cluster count for cluster: " + clusterName +
              ", scheme: " + scheme + ", partition: " + partitionId + ", exception: " + e);
      return -1;
    }
```